observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -187,10 +187,11 @@ public Predicate<String> getSubsystemFilter() {
         }, OperationContext.Stage.VERIFY);
 
         return CapabilityServiceInstaller.builder(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY, collectorSupplier)
-                .requires(List.of(mccf, executor, processStateNotifier))
-                .withCaptor(captor::set) // capture the provided value
-                .asActive() // Start actively
-                .build();
+            .requires(List.of(mccf, executor, processStateNotifier))
+            .withCaptor(captor::set) // capture the provided value
+            .onRemove(wildFlyRegistry::close)
```

Comment:
@pferraro Would appreciate you chiming in on this to make sure it's the intended use of the API.

Reply:
Moreover, this should be onStop(..) instead.

Reply:
onRemove(...) should conclude actions initiated by service installation.
onStop(...) should conclude actions initated by service start.

Reply:
Just to elaborate on my above comment, I am under suspicion, that the collector should NOT start pushing metrics on service installation, but rather than on actual service start, and thus analogously the stop should be onStop. Unless there are reasons that I am not seeing now?

For example: in JGroups channel service case, the provider provides a JChannel, onStart the channel is connected and onStop the channel is disconnected.

Reply:
I agree with @rhusar here, but I think we may need to defer that the WF 36 as this is a 35 Blocker and needs to be ready for merge in 2 days.  This is convoluted but seems to work.

It's not WildFlyCollector that is pushing metrics on service installation; it's WildFlyOtlpRegistry that is pushing metrics as soon as this configure() method instantiates it.

I think WildFlyRegistry, JmxMicrometerCollector and MicrometerCollector are services, with WildFlyRegistry injected into the others via a Supplier. It's the builder for the WildFlyRegistry service that gets .onStop(Functions::closingConsumer). 

Or simpler they all go in one service that instantiates the WildFlyRegistry and the JmxMicrometerCollector.

In the old code WildFlyOtlpRegistry was constructed (and thus started publishing) in a service start.

Reply:
Agreed. As I was working through this, there were several things I saw that I wanted to clean up, but I was trying to minimize the changes, given the timeline for this. I have filed a JIRA for this: https://issues.redhat.com/browse/WFLY-20148